test/unit/test-new-kernel-api.cpp

```diff
@@ -0,0 +1,605 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```

Comment:
Would this file and tests inside it be named better if it were something like test-kernel-lambda-args.cpp?? In other words break out the tests that do contortions of the lambda arguments into a separate test source file.

Reply:
Yup, I just renamed the file. The new source file just looks at testing the different API.